### PR TITLE
Version 2.5.0

### PR DESCRIPTION
**Pull request by mpeyfuss**
Added batch mint functionality that does not use ERC-2309 in `ERC721TL` and `Shatter`.

## ERC721TL
- migrated to using `Transfer` events on the `batchMint` function
- created a `batchMintUltra` function that can be used if the target platform listens to ERC-2309's `ConsecutiveTransfer` event. This should not be the default function to use.

## Shatter
- migrated to using `Transfer` events on the `shatter` function
- created a `shatterUltra` function that uses ERC-2309's `ConsecutiveTransfer` event. This should not be the default function to use.

Tests have been updated to reflect this and 1024 fuzz runs were completed with passing results. (minor updates included to the test suite to ensure passing results).

Updated gas cost for `batchMint` on `ERC721TL`:
| NFTs | Gas Cost |
|------|-----------|
| 5 | 198,000 |
| 10 | 209,000 |
| 100 | 402,000 |
| 200 | 616,000 |
| 500 | 1,259,000 |
| 1000 | 2,331,000 |

### CODE REVIEW

**Comment by koloz193** [comment]
What’s the gas cost of these batch mints before removing?

I think that we should be at the forefront of what’s best vs what’s most supported. Especially since this is a web2 integration that breaks and can easily be updated.

File: src/core/ERC721TL.sol [hunk 1]
```diff
@@ -104,11 +106,6 @@ contract ERC721TL is
                                 Events
     //////////////////////////////////////////////////////////////////////////*/
 
-    /// @dev This event is for consecutive transfers per EIP-2309
```
Comment (bstrauss23) [inline src/core/ERC721TL.sol:107]: Aren't we keeping 2309 in the contract, just as an alternative option?

**Review by bstrauss23 — APPROVED** [review]
Looks good as far as my understanding goes! I trust your tests more than my own eyes.

**Comment by mpeyfuss** [comment]
> What’s the gas cost of these batch mints before removing?
> 
> I think that we should be at the forefront of what’s best vs what’s most supported. Especially since this is a web2 integration that breaks and can easily be updated.

gas cost is included in the description of this PR

File: src/core/ERC721TL.sol (hunk 1 above)
Comment (mpeyfuss) [reply src/core/ERC721TL.sol:107]: Resorted to importing it from OpenZeppelin rather than manually defining.

**Pull request merged**